### PR TITLE
Adjust Doxygen configuration

### PR DESCRIPTION
**Pull request by klecki**
## Description:
Extracted from #4071.

Replace the usage of `///` with `/** */` which is used
across the project.

Unify the usage of grouping directives.
The header should use either just `/** @{ */`
or additionaly @name or @defgroup.

  @name can be used for member functions, groups them,
  and keeps them inlined on class page.

  @defgroup can be used with free function or friend
  definitions, or other top level entities. It generates
  a dedicated page for the group.

The foother should use `/** @} */`

Set DISTRIBUTE_GROUP_DOC = YES - now the first docstring
in a group is replicated in the other members that didn't
provide a docstring.

Set EXTRACT_ALL = YES - now, all entities are automatically
documented - otherwise, for example, free functions were
ignored in file without @file documentation.
This enables most free-functions (for example C API)
and not just Classes and their members to be documented.

Set EXTRACT_PRIVATE = YES - 
The DALI C++ API is an internal backend API,
and the private functions tend to be documented
to aid the developers and the development.
It should be rendered into HTML version,
the method get the dedicated Private section.

### Affected modules and functionalities:
Docs

### Key points relevant for the review:
:eyes: 

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [X] Documentation updated
  - [X] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Review by awolant — APPROVED** [review]

File: dali/operators/image/remap/warp.h [hunk 1]
```diff
@@ -229,10 +229,10 @@ class Warp : public SequenceOperator<Backend> {
   const OpSpec &Spec() const { return this->spec_; }
 
  private:
-  /// @defgroup WarpStaticType Dynamic to static type routing
+  /** @defgroup WarpStaticType Dynamic to static type routing */
```
Comment (jantonguirao) [inline dali/operators/image/remap/warp.h:232]: Shouldn't this be a single `/**` and `*/` for the group of lines?

**Review by jantonguirao — APPROVED** [review]

File: dali/operators/image/remap/warp.h (hunk 1 above)
Comment (klecki) [reply dali/operators/image/remap/warp.h]: Done. Actually, it's in class, so it should be named group

**Comment by klecki** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**